packages/calcite-components/src/components/date-picker-month-header/date-picker-month-header.tsx

```diff
@@ -279,9 +340,33 @@ export class DatePickerMonthHeader {
   /*
    * Update active month on clicks of left/right arrows
    */
-  private handleArrowClick = (event: MouseEvent | KeyboardEvent, date: Date): void => {
+  private handleArrowClick = async (
+    event: MouseEvent | KeyboardEvent,
+    date: Date,
+  ): Promise<void> => {
     event.preventDefault();
-    this.calciteInternalDatePickerSelect.emit(date);
+
+    await this.handlePenultimateValidMonth(event);
+    this.calciteInternalDatePickerMonthHeaderSelect.emit(date);
+  };
+
+  private handleMonthChange = (event: CustomEvent): void => {
+    const target = event.target as HTMLCalciteOptionElement;
+    const { abbreviated, wide } = this.localeData.months;
+    const localeMonths = this.monthStyle === "wide" ? wide : abbreviated;
+    const monthIndex = localeMonths.indexOf(target.value);
+    let newDate = getDateInMonth(this.activeDate, monthIndex);
+    if (!inRange(newDate, this.min, this.max)) {
+      newDate = dateFromRange(newDate, this.min, this.max);
+    }
+    this.calciteInternalDatePickerMonthHeaderSelect.emit(newDate);
+    this.setSelectMenuWidth(this.monthPickerEl);
+  };
+
+  private handleKeyDown = (event: KeyboardEvent): void => {
+    if (event.key === "ArrowDown" || event.key === "ArrowUp") {
+      event.stopPropagation();
```

Comment:
FIXME: let's avoid stopping the propagation of native events.

Reply:
This is intentional to avoid closing `date-picker` in `input-date-picker` on ArrowKeys.

Reply:
We cancel the event for these cases. `input-date-picker` will [ignore canceled arrow key presses](https://github.com/Esri/calcite-design-system/blob/anveshmekala/3455-add-month-and-year-picker/packages/calcite-components/src/components/input-date-picker/input-date-picker.tsx#L943-L945).